### PR TITLE
chore: add back scaling factor choice

### PR DESCRIPTION
**Pull request by PaddyMc**
## What is the purpose of the change

This PR migrates the selected cl pool incentives to the new scaling factor, in this PR we add back in the code path choice for selected pool scaling factors

currently the pool added to this migration are:

```
// MigratedIncentiveAccumulatorPoolIDsV24 is a map that defines pools to migrate to the latest scalingFactor for v24.
var MigratedIncentiveAccumulatorPoolIDsV24 = map[uint64]struct{}{
	// token0 eth
	// token1 uosmo
	// liquidity 272k
	1281: {},
	// token0 usdt
	// token1 dia
	// liquidity 500
	1276: {},
	// token0 usdc
	// token1 dai
	// liqudity 1.55k
	1275: {},
	// token0 usdc
	// token1 dai
	// liqudity 52.9k
	1260: {},
}
```

### How to test

To test this PR you need to use the `in-place-testnet` 

Download the latest snapshot and genesis

```bash
OSMOSIS_HOME=$HOME/.osmosisd

SNAPSHOT_URL=$(curl -sL https://snapshots.osmosis.zone/latest)
RPC_URL=https://rpc.osmosis.zone
ADDRBOOK_URL="https://rpc.osmosis.zone/addrbook"
GENESIS_URL=https://github.com/osmosis-labs/osmosis/raw/main/networks/osmosis-1/genesis.json

rm -rf $HOME/.osmosisd/

osmosisd init test --chain-id testing --home=$HOME/.osmosisd

# Copy genesis
echo -e "\nDownloading genesis file..."
wget $GENESIS_URL -O $OSMOSIS_HOME/config/genesis.json
echo ✅ Genesis file downloaded successfully.

# Download latest snapshot
echo -e "\nDownloading latest snapshot..."
wget -O - $SNAPSHOT_URL | lz4 -d | tar -C $OSMOSIS_HOME/ -xf -
echo -e ✅ Snapshot downloaded successfully.

# Run the node
osmosisd start --home=$HOME/.osmosisd
```

You need to install v23.0.8 to run the following command
```
osmosisd in-place-testnet edgenet osmo12smx2wdlyttvyzvzg54y2vnqwq2qjateuf7thj --trigger-testnet-upgrade v24
```

Wait until the node stops, build this branch and run
```
osmosisd start
```

This will run the v24 migration and migrate selected CL pool incentive accumlators

How to test manually:

- Create an incentive for a pool
- Create a CL position in a tight range
- Swap tokens
- Check that rewards have been distributed

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality to selectively migrate CL incentives, enhancing the migration process.
	- Added a feature to retrieve and apply incentive scaling factors for pools, improving the accuracy of incentive calculations.

- **Refactor**
	- Updated references and variable names related to migrated incentive accumulator pool IDs across multiple upgrade paths.
	- Adjusted logic for managing pool IDs during migration to ensure compatibility with new incentive scaling features.

- **Tests**
	- Added and updated tests to verify the correct application of incentive scaling factors and the selective migration of incentives.

### CODE REVIEW

File: x/concentrated-liquidity/incentives.go [hunk 1]
```diff
@@ -1101,6 +1112,36 @@ func (k Keeper) getLargestAuthorizedUptimeDuration(ctx sdk.Context) time.Duratio
...
+	}
+
+	// If the given pool ID is in the migrated incentive accumulator pool IDs, we return the perUnitLiqScalingFactor.
+	_, isMigrated := types.MigratedIncentiveAccumulatorPoolIDs[poolID]
+	if isMigrated {
+		return perUnitLiqScalingFactor, nil
+	}
+
+	// If the given pool ID is in the migrated incentive accumulator pool IDs (v24), we return the perUnitLiqScalingFactor.
+	_, isMigrated = types.MigratedIncentiveAccumulatorPoolIDsV24[poolID]
```
Comment (PaddyMc) [inline x/concentrated-liquidity/incentives.go:1136]: This is the most important line of this PR IMO, now we have a second check to return scaling factors depending on the v24 migration map

File: CHANGELOG.md [hunk 2]
```diff
@@ -82,6 +82,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * [#7855](https://github.com/osmosis-labs/osmosis/pull/7855) Whitelist address parameter for setting fee tokens
 * [#7857](https://github.com/osmosis-labs/osmosis/pull/7857) SuperfluidDelegationsByValidatorDenom query now returns equivalent staked amount
 * [#7912](https://github.com/osmosis-labs/osmosis/pull/7912) Default timeoutCommit to 2s
+* [#7951](https://github.com/osmosis-labs/osmosis/pull/7951) Only migrate selected cl incentives
```
Comment (czarcas7ic) [inline CHANGELOG.md:85]: Need to remove the changelog entry on line 78

**Review by czarcas7ic — APPROVED** [review]
This LGTM, but certainly should have a second ACK given how serious the effects of implementing this wrong would be

**Review by p0mvn — APPROVED** [review]

**Comment by p0mvn** [comment]
How long did migration take now?

File: CHANGELOG.md (hunk 2 above)
Comment (PaddyMc) [reply CHANGELOG.md:84]: Removed, good catch

File: x/concentrated-liquidity/incentives_test.go [hunk 3]
```diff
@@ -3735,6 +3735,64 @@ func (s *KeeperTestSuite) TestComputeTotalIncentivesToEmit() {
...
+	s.Require().NotZero(len(types.MigratedIncentiveAccumulatorPoolIDsV24))
+
+	var exampleOverwritePoolID uint64
+	for poolID := range types.MigratedIncentiveAccumulatorPoolIDs {
+		exampleOverwritePoolID = poolID
+		break
+	}
+
+	var exampleOverwritePoolIDv24 uint64
+	for poolIDv24 := range types.MigratedIncentiveAccumulatorPoolIDs {
```
[the remark this answers is not in the document]
Comment (PaddyMc) [reply x/concentrated-liquidity/incentives_test.go]: Thank you, sorry this was a bad mistake, updated :bow:

**Comment by PaddyMc** [comment]
> How long did migration take now?

less than 10 seconds, posted a video internally in slack

**Pull request merged**